### PR TITLE
Breaking change API Group, Labels and Annotations

### LINKED ISSUE #662: Naming rule for annotations and labels
Current `cosmo/RESOURCE-KEY`

↓

New `RESOURCE.DOMAIN/KEY`

e.g.

Current  `cosmo/ws-deployment`

↓

New `workspace.cosmo-workspace.github.io/deployment-name`

### PR DESCRIPTION
**Pull request by jlandowner**
Close #662

1. API Group

`cosmo.cosmo-workspace.github.io` and `workspace.cosmo-workspace.github.io` are integrated to `cosmo-workspace.github.io`

 | After | Before |
|:---|:---|
|  `templates.cosmo-workspace.github.io` |`templates.cosmo.cosmo-workspace.github.io` | 
|  `instances.cosmo-workspace.github.io` |`instances.cosmo.cosmo-workspace.github.io` | 
|  `clustertemplates.cosmo-workspace.github.io` |`clustertemplates.cosmo.cosmo-workspace.github.io` | 
|  `clusterinstances.cosmo-workspace.github.io` |`clusterinstances.cosmo.cosmo-workspace.github.io` | 
|   `users.cosmo-workspace.github.io` |`users.workspace.cosmo-workspace.github.io` |
|  `workspaces.cosmo-workspace.github.io` |`workspaces.workspaces.cosmo-workspace.github.io` | 

2. Labels

| Resource | After |Before | 
|:---|:---|:---|
| (instance-owned resources) |  `cosmo-workspace.github.io/template` |`cosmo/template` | 
| (instance-owned resources) |   `cosmo-workspace.github.io/instance` |`cosmo/instance` |
| Namespace  | `cosmo-workspace.github.io/user` |`cosmo/user-name` | 
| Template | `cosmo-workspace.github.io/type` |`cosmo/type` | 
| (TemplateLabelEnumTypeUserAddon)  | `useraddon` |`user-addon` | 

3. Annotations

| Resource | After |Before | 
|:---|:---|:---|
| Template  | `cosmo-workspace.github.io/disable-nameprefix` |`cosmo/disable-nameprefix` | 
| Template  |  `cosmo-workspace.github.io/skip-validation` |`cosmo/skip-validation` |
| Template | `workspace.cosmo-workspace.github.io/urlbase` |`cosmo/ws-urlbase` | 
| Template |  `workspace.cosmo-workspace.github.io/deployment` |`cosmo/ws-deployment` |
| Template | `workspace.cosmo-workspace.github.io/service` |`cosmo/ws-service` | 
| Template | `workspace.cosmo-workspace.github.io/ingress` |`cosmo/ws-ingress` | 
| Template | `workspace.cosmo-workspace.github.io/service-main-port` |`cosmo/ws-service-main-port` | 
| Template |  `useraddon.cosmo-workspace.github.io/default` |`cosmo/default-user-addon` |
| Secret | `cosmo-workspace.github.io/default-password` |`cosmo/default` |

### CODE REVIEW

File: internal/authproxy/suite_test.go [hunk 1]
```diff
@@ -67,7 +66,7 @@ var _ = BeforeSuite(func() {
 
 	_ = clientgoscheme.AddToScheme(scheme)
 	_ = cosmov1alpha1.AddToScheme(scheme)
-	_ = wsv1alpha1.AddToScheme(scheme)
+	_ = cosmov1alpha1.AddToScheme(scheme)
```
Comment (oruharo) [inline internal/authproxy/suite_test.go:69]: This line is duplicated.

File: internal/cmd/netrule_test.go [hunk 2]
```diff
@@ -40,7 +39,7 @@ var _ = Describe("cosmoctl [netrule]", func() {
 		scheme := runtime.NewScheme()
 		_ = clientgoscheme.AddToScheme(scheme)
 		_ = cosmov1alpha1.AddToScheme(scheme)
-		_ = wsv1alpha1.AddToScheme(scheme)
+		_ = cosmov1alpha1.AddToScheme(scheme)
```
Comment (oruharo) [inline internal/cmd/netrule_test.go:42]: This line is duplicated.

File: internal/cmd/suite_test.go [hunk 3]
```diff
@@ -44,7 +43,7 @@ const DefaultURLBase = "https://{{NETRULE_GROUP}}-{{INSTANCE}}-{{USER_NAME}}.dom
 
 func init() {
 	cosmov1alpha1.AddToScheme(scheme.Scheme)
-	wsv1alpha1.AddToScheme(scheme.Scheme)
+	cosmov1alpha1.AddToScheme(scheme.Scheme)
```
Comment (oruharo) [inline internal/cmd/suite_test.go:46]: This line is duplicated.

File: example/code-server-ingress-no-storage.yaml [hunk 4]
```diff
@@ -141,7 +138,6 @@ spec:
       name: '{{INSTANCE}}-code-server'
       namespace: '{{NAMESPACE}}'
     spec:
-      ingressClassName: '{{INGRESS_CLASS}}'
```
[the remark this answers is not in the document]
Comment (jlandowner) [reply example/code-server-ingress-no-storage.yaml:144]: No deep meaning. It's just for easy applying to the local-run-test environment

**Comment by jlandowner** [comment]
Fixed

File: example/code-server-ingress-no-storage.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlandowner) [reply example/code-server-ingress-no-storage.yaml:144]: Is it better to revert?

[the remark this answers is not in the document]
Comment (oruharo) [reply example/code-server-ingress-no-storage.yaml:144]: https://kubernetes.io/docs/concepts/services-networking/ingress/
> If the ingressClassName is omitted, a [default Ingress class](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class) should be defined.
> There are some ingress controllers, that work without the definition of a default IngressClass. For example, the Ingress-NGINX controller can be configured with a [flag](https://kubernetes.github.io/ingress-nginx/#what-is-the-flag-watch-ingress-without-class) --watch-ingress-without-class. It is [recommended](https://kubernetes.github.io/ingress-nginx/#i-have-only-one-instance-of-the-ingresss-nginx-controller-in-my-cluster-what-should-i-do) though, to specify the default IngressClass as shown [below](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class).

After reading this, I thought it was best not to omit it.
I will leave it to you.

[the remark this answers is not in the document]
Comment (jlandowner) [reply example/code-server-ingress-no-storage.yaml]: I reverted.

FYI local-run-test  environment does not have any IngressClasses.

```sh
coder@test-env:local-run-test$ k get ingressclass  -A
No resources found
```

**Comment by jlandowner** [comment]
Fixed

**Pull request merged**